src/main/java/se/liu/ida/hefquin/engine/queryplan/logical/LogicalPlanCounter.java

```diff
@@ -0,0 +1,14 @@
+package se.liu.ida.hefquin.engine.queryplan.logical;
+
+public class LogicalPlanCounter {
+	private static int counter = 1;  
+    protected int operatorID;  
+
+    public LogicalPlanCounter() {
+        this.operatorID = counter++;  
+    }
+
+    public int getOperatorID() {
```

Comment:
Rename this function to `getID`, and then add it also to the `LogicalOperator` interface (and then add the `@Override` annotation here).